stackstate_checks_base/tests/test_http_helper.py

```diff
@@ -0,0 +1,458 @@
+        assert HTTPHelperResponseHandler({
+            "response_type_validation": HTTPRequestType.JSON
+        }).validate_body_type(valid_response)
+        try:
+            HTTPHelperResponseHandler({
+                "response_type_validation": HTTPRequestType.JSON
+            }).validate_body_type(invalid_response)
+            assert False
+        except ValueError:
+            assert True
+
+
+class TestHTTPHelperBase(unittest.TestCase):
+    def test_full_get(self):
+        http = HTTPHelper()
+        response = http.get({
+            "endpoint": "mock://test.com",
+            "mock_enable": True,
+            "mock_status": 200,
+            "mock_response": {
+                'hello': 'world',
+                'pong': True,
+            },
+            "auth_data": {
+                'username': 'world',
+                'password': 'hello'
+            },
+            "auth_type": HTTPRequestType.JSON,
+            "session_auth_data": {
+                'username': 'world',
+                'password': 'hello'
+            },
+            "session_auth_type": HTTPRequestType.JSON,
+            "body": {"hello": "world", "pong": True},
+            "proxy": {"hello": "world"},
+            "headers": {"a": "1"},
+            "session_headers": {"b": "1"},
+            "query": {"c": "1"},
+            "timeout": 30,
+            "ssl_verify": True,
+            "retry_policy": {},
+            "request_schematic_validation": BodyRequestSchematicTest,
+            "request_type_validation": HTTPRequestType.JSON,
+            "response_status_code_validation": 200,
+            "response_type_validation": HTTPResponseType.JSON,
+            "response_schematic_validation": BodyResponseSchematicTest,
+        })
+
+        assert response.request.method == "GET"
+        assert response.status_code == 200
+        assert json.loads(response.content.decode(response.encoding)).get("hello") == "world"
+        assert response.request.url == "mock://test.com"
+        assert response.request.headers == {'a': '1',
+                                            'Content-Length': '21',
+                                            'Content-Type': 'application/x-www-form-urlencoded'}
+
+    def test_compact_get(self):
+        http = HTTPHelper()
+        response = http.get({
+            "endpoint": "mock://test.com",
+            "mock_enable": True,
+            "mock_status": 201,
+            "mock_response": {
+                'hello': 'world',
+                'pong': True,
+            },
+        })
+
+        assert response.request.method == "GET"
+        assert response.status_code == 201
+        assert json.loads(response.content.decode(response.encoding)).get("hello") == "world"
+        assert response.request.url == "mock://test.com"
+
+    def test_full_unicode_request_and_response(self):
+        unicode = ""
+        for i in range(100):
+            unicode = unicode + chr(i)
```

Comment:
How is this testing unicode? You are just setting `unicode` to a string containing the first 100 ASCII characters. You are not even getting outside the range of 7 bit ASCII this way.

Pleas read this: https://www.joelonsoftware.com/2003/10/08/the-absolute-minimum-every-software-developer-absolutely-positively-must-know-about-unicode-and-character-sets-no-excuses/

The bare minimum you should test for is receiving responses that contain some weird Chinese characters in a non UTF-8 encoding, which you should be able to decode properly by reading the `Content-Type` HTTP header. And by you I actually mean requests; that should all be taken care of by requests if all goes well.